src/v2/providers/pubsub.ts

```diff
@@ -353,3 +356,70 @@ export function onMessagePublished<T = any>(
 
   return func;
 }
+
+/**
+ * @internal
+ *
+ * Adds a v1-style context to the event.
```

Comment:
nit* this is adding both context and data (`message`) to it. Maybe we need a slightly better name + update the docstring to reflect that?